libfreerdp/core/freerdp.c

```diff
@@ -627,7 +627,11 @@ UINT32 freerdp_error_info(freerdp* instance)
 	return instance->context->rdp->errorInfo;
 }
 
-void freerdp_set_error_info(rdpRdp* rdp, UINT32 error) {
+void freerdp_set_error_info(rdpRdp* rdp, UINT32 error)
+{
+	if (!rdp)
+		return;
```

Comment:
@mfleisz you're right with the public API part, but ```rdp_set_error_info``` has a return value, which should reflect such misuses.